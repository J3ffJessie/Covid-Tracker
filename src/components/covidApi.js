import React, { useEffect, useState, useRef } from "react"
import { Link } from "gatsby"
import axios from "axios"
import states from "../utils/states"
import NumberFormat from "react-number-format"

import Layout from "./layout"
import SEO from "./seo"

export default function ApiCall() {
  const firstUpdate = useRef(false)
  const [requestState, setRequestState] = useState("idle")

  const [data, setData] = useState("")

  const [state, setState] = useState("al")

  const onStateSelect = e => {
    e.preventDefault()
    setState(e.target.value)
  }

  async function fetchCovidData(value) {
    const response = await axios.get(
      `https://api.covidtracking.com/v1/states/${value}/current.json`
    )
    setData(response.data)
  }

  useEffect(() => {
    fetchCovidData(state)
  }, [state])

  return (
    <div>
      <div className="data">
        <div className="banner">
          <div className="drop-down">
            <select
              name="state"
              id="stateSelect"
              onChange={e => onStateSelect(e)}
            >
              {states.map(s => (
                <option key={s.value} value={s.value}>
                  {s.label}
                </option>
              ))}
            </select>
          </div>
        </div>
      </div>
      <h1>Covid Data</h1>
      <details open>
        <summary>Show Statistics</summary>
        <div className="container">
          <table>
            <thead>
              <tr>
                <th className="table-title">Category</th>
                <th className="table-title">{data.state}'s Statistics </th>
              </tr>
            </thead>
            <tbody>
              {data && data.lastUpdateEt ? (
                <tr>
                  <td>Date/Time of Last Update</td>
                  <td>{data.lastUpdateEt}</td>
                </tr>
              ) : null}
              {data && data.positive ? (
                <tr>
                  <td>Positive Results</td>
                  <td>
                    <NumberFormat
                      value={data.positive}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.negative ? (
                <tr>
                  <td>Negative Results</td>
                  <td>
                    <NumberFormat
                      value={data.negative}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.totalTestResults ? (
                <tr>
                  <td>Total Test Results</td>
                  <td>
                    <NumberFormat
                      value={data.totalTestResults}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.hospitalizedCurrently ? (
                <tr>
                  <td># of Patients Currently Hospitalized</td>
                  <td>
                    <NumberFormat
                      value={data.hospitalizedCurrently}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.inIcuCurrently ? (
                <tr>
                  <td># of Patients in ICU Currently</td>
                  <td>
                    <NumberFormat
                      value={data.inIcuCurrently}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.positiveIncrease ? (
                <tr>
                  <td>Increase in Positive Results from Last Update</td>
                  <td>
                    <NumberFormat
                      value={data.positiveIncrease}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.negativeIncrease ? (
                <tr>
                  <td>Increase in Negative Results from Last Update</td>
                  <td>
                    <NumberFormat
                      value={data.negativeIncrease}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.deathIncrease ? (
                <tr>
                  <td># of Deaths since Last Update</td>
                  <td>
                    <NumberFormat
                      value={data.deathIncrease}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
              {data && data.hospitalizedIncrease ? (
                <tr>
                  <td># of Increased Hospitalizations</td>
                  <td>
                    <NumberFormat
                      value={data.hospitalizedIncrease}
                      displayType={"text"}
                      thousandSeparator={true}
                    />
                  </td>
                </tr>
              ) : null}
            </tbody>
          </table>
        </div>
      </details>
      {/* <details className="chart-container">
          <summary>Show Charts</summary>
          <div className="charts"></div>
        </details> */}
    </div>
  )
}
