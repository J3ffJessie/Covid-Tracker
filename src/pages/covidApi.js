import React, { useState } from "react"
import { Link } from "gatsby"
import axios from 'axios'

import Layout from "../components/layout"
import SEO from "../components/seo"


const defaultData = {
date: 20210126,
positive: 0,
negative: 0 ,
totalTestResults: 0,
hospitalizedCurrently:0,
inIcuCurrently: 0,
lastUpdateEt:0,
positiveIncrease:0,
negativeIncrease: 0,
total: 0,
totalTestResultIncrease: 0,
deathIncrease: 0,
hospitalizedIncrease: 0
}

export function ApiCall() {


  const [requestState, setRequestState] = useState('idle')

  const [data, setData] = useState({defaultData})

  // const [userState, setUserState] = useState('')

  React.useEffect( () => {

    setRequestState('pending')
    //api call
    axios.get('https://api.covidtracking.com/v1/states/in/current.json')
    .then((response) => {
      setRequestState('success')
      setData(response.data)
      console.log(data);
    })
    .catch((error) => {
      setRequestState('failure')
      console.error(error)
    })
  }, [])


  return(
    <div>
      <div className='data'>
        <div className="banner">
          <div className="container">
          <h1>Covid Data</h1>
          <p>Here are the stats for the State chosen: </p>
          </div>
        </div>

        {requestState === 'pending' && <h2>Loading......</h2>}

        {requestState === 'failure' && <h2>Sorry, the Data is not available at this time.</h2>}

        {requestState === 'success' && (() => {
          return(
        <div>
          <h1>{data.date}</h1>
          <p>{data.positive}</p>
          <p>{data.negative}</p>
          <p>{data.totalTestResults}</p>
          <p>{data.hospitalizedCurrently}</p>
          <p>{data.inIcuCurrently}</p>
          <p>{data.lastUpdateEt}</p>
          <p>{data.total}</p>
          <p>{data.totalTestResultIncrease}</p>
          <p>{data.positiveIncrease}</p>
          <p>{data.negativeIncrease}</p>
          <p>{data.deathIncrease}</p>
          <p>{data.hospitalizedIncrease}</p>
        </div>

      )
    })}
      </div>
    </div>
  )
}


export default ApiCall 

// const SecondPage = () => (
//   <Layout>
//     <SEO title="Page two" />
//     <h1>Hi from the second page</h1>
//     <p>Welcome to page 2</p>
//     <Link to="/">Go back to the homepage</Link>
//   </Layout>
// )

// export default SecondPage
