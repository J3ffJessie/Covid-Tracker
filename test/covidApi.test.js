import React from 'react';
import { act } from 'react-dom/test-utils';
import covidApi from '../src/components/covidApi';

let container = null
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting testing
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("renders covid statistics", async () => {
    const fakeData = {
        state: 'BS',
        date: '2020201',
        positive: '13',
        negative: '0',
        totalTestResults: '45',
        hostpitalizedCurrently: '0',
        inIcuCurrently: '0',
        lastUpdateEt: '2/1/2021 01:00',
        positiveIncrease: '10',
        negativeIncrease: '0',
        deathIncrease: '0',
        hospitalizedIncrease: '0'
    };
    jest.spyOn(global, "fetch").mockImplementation(() => 
    Promise.resolve({
        json: () => Promise.resolve(fakeData)
    }));

    expect(container.querySelector("p").textContent).toBe(fakeData.state);
    expect(container.querySelector("p").textContent).toBe(fakeData.date);
    expect(container.querySelector("p").textContent).toBe(fakeData.positive);
    expect(container.querySelector("p").textContent).toBe(fakeData.negative);
    expect(container.querySelector("p").textContent).toBe(fakeData.totalTestResults);
    expect(container.querySelector("p").textContent).toBe(fakeData.hostpitalizedCurrently);
    expect(container.querySelector("p").textContent).toBe(fakeData.inIcuCurrently);
    expect(container.querySelector("p").textContent).toBe(fakeData.lastUpdateEt);
    expect(container.querySelector("p").textContent).toBe(fakeData.positiveIncrease);
    expect(container.querySelector("p").textContent).toBe(fakeData.negativeIncrease);
    expect(container.querySelector("p").textContent).toBe(fakeData.deathIncrease);
    expect(container.querySelector("p").textContent).toBe(fakeData.hospitalizedIncrease);
    

    global.fetch.mockRestore();
});


