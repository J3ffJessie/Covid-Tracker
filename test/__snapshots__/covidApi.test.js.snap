// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders covid statistics loading state 1`] = `
<DocumentFragment>
  loading....
</DocumentFragment>
`;
